hollow/src/main/java/com/netflix/hollow/api/producer/HollowProducerListenerV2.java

```diff
@@ -0,0 +1,36 @@
+/*
+ *
+ *  Copyright 2018 Netflix, Inc.
+ *
+ *     Licensed under the Apache License, Version 2.0 (the "License");
+ *     you may not use this file except in compliance with the License.
+ *     You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *     Unless required by applicable law or agreed to in writing, software
+ *     distributed under the License is distributed on an "AS IS" BASIS,
+ *     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *     See the License for the specific language governing permissions and
+ *     limitations under the License.
+ *
+ */
+package com.netflix.hollow.api.producer;
+
+/**
+ * An extension of {@link HollowProducerListener} to allow adding new methods without
+ * breaking backwards compatability.
+ * Subclasses should extend {@link NoOpHollowProducerListener} to avoid having to override
+ * every method in this interface.
+ * TODO(hollow3): Collapse this into HollowProducerListener.
+ */
+public interface HollowProducerListenerV2 extends HollowProducerListener {
```

Comment:
I hate `V{N}` in class names, yet this is probably the best way to address this. Making a new name like `EnhancedHollowProducerListener` falls short when you need to enhance it again.

Reply:
Yeah agreed, not ideal, but works for our use case. In theory, in the next version of hollow this file will go away.